cub/cub/agent/single_pass_scan_operators.cuh

```diff
@@ -122,9 +124,27 @@ enum ScanTileStatus
   SCAN_TILE_INCLUSIVE, // Inclusive tile prefix is available
 };
 
+/**
+ * Indicates whether to use a st.release when writing the partial or inclusive tile state (EnforceStoreRelease::yes) or
+ * not.
+ */
+enum class EnforceStoreRelease
+{
+  no,
+  yes
+};
+
 namespace detail
 {
 
+template <EnforceStoreRelease StoreRelease>
+using store_release_tag_t =
+  ::cuda::std::conditional<(StoreRelease == EnforceStoreRelease::no),
+                           Int2Type<static_cast<int>(EnforceStoreRelease::no)>,
+                           Int2Type<static_cast<int>(EnforceStoreRelease::yes)>>;
+using store_release_tag_no  = store_release_tag_t<EnforceStoreRelease::no>;
+using store_release_tag_yes = store_release_tag_t<EnforceStoreRelease::yes>;
+
```

Comment:
Why do we need this dance again? We could pass the enum value just fine couldnt we

Reply:
This was some machinery used for tag-dispatching. I switched the implementation to use SFINAE on the relevant member function templates.